scipy/_lib/_util.py

```diff
@@ -337,3 +338,69 @@ def getargspec_no_self(func):
         if argspec.args[0] == 'self':
             argspec.args.pop(0)
         return argspec
+
+
+class PoolWrapper(object):
+    """
+    Wrapper for working with objects that have map-like methods, such as
+    `multiprocessing.Pool`. Used for parallelisation.
+
+    Parameters
+    ----------
+    pool : int or map-like object
+        If `pool` is an `int` then it specifies the number of threads to
+        use for parallelization. If `pool == 1`, then no parallel
+        processing is used and the map builtin is used. If `pool == 0` then
+        the pool will utilise all available CPU.
+        If pool is an object with a map method that follows the same
+        calling sequence as the built-in map function, then this pool is
+        used for parallelisation.
+    """
+    def __init__(self, pool=1):
+        self.pool = None
+        self._mapfunc = map
+        self._own_pool = False
+
+        if hasattr(pool, 'map'):
+            self.pool = pool
+            self._mapfunc = self.pool.map
+        else:
+            # user supplies a number
+            if int(pool) == 0:
```

Comment:
This should be -1 I think, for consistency with `cKDTree` and `scikit-learn`.

Reply:
In that case which of `workers=0` or `workers=1` represents the serial cases? And what does the other represent?

Reply:
But I guess for consistency reasons it has to be changed.

Reply:
the default (1). nothing else needs to change. I think in general, -1 for "all/auto" is probably more common than 0. E.g. lapack lwork=-1 queries.